### PR TITLE
Consolidated High Fidelity PR installer

### PR DESCRIPTION
**Pull request by murillodigital**
1) Both interface as well as stack + stack manager are bundled in a single installer, all using the same copy of Qt and living in the same High Fidelity Directory --- currently the user has no ability to select which components are wanted, that will change.
2) Stack Manager will look for local stack, all components match the same built commit.
3) CPack is used to automate the installer generation, a new PACKAGE project is created for PR builds - this can be used by any developer to go through entire release process. (Requires NSIS installed on system)

There are a few things that still need fixing / enhancing which will come on further PRs.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e186409221b2adbc9ace33c1b7101796ca63a71 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e186409221b2adbc9ace33c1b7101796ca63a71 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by murillodigital** [comment]
Please use the following link for testing - applying this to the pipeline requires target changes in build jobs and installer name changes:

http://s3.amazonaws.com/hifi-public/pr-builds/High Fidelity-PR.6497-win32.exe

**Comment by imgntn** [comment]
@murillodigital just tested this on windows 10 and everything seemed to install fine -- only minor suggestion would be to start the installation in the "High Fidelity Development" folder by default, like other PR builds, otherwise the Program Files folder might get a little unruly.  But if that's for some reason difficult, it isn't too hard for the user to manually change it.

File: stack-manager/src/AppDelegate.cpp [hunk 1]
```diff
@@ -115,14 +111,15 @@ AppDelegate::AppDelegate(int argc, char* argv[]) :
...
-
     _checkVersionTimer.setInterval(0);
     connect(&_checkVersionTimer, SIGNAL(timeout()), this, SLOT(checkVersion()));
     _checkVersionTimer.start();
 
     connect(this, &QApplication::aboutToQuit, this, &AppDelegate::stopStack);
+
+	_window->setRequirementsLastChecked(QDateTime::currentDateTime().toString());
+	_window->show();
+	toggleStack(true);
```
Comment (AndrewMeadows) [inline stack-manager/src/AppDelegate.cpp]: The indentation here is funny... using tabs instead of spaces.

File: stack-manager/src/GlobalData.cpp [hunk 2]
```diff
@@ -32,11 +32,8 @@ GlobalData::GlobalData() {
     _resourcePath = "resources/";
     _assignmentClientExecutable = "assignment-client";
     _domainServerExecutable = "domain-server";
-    QString applicationSupportDirectory = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
-    if (PR_BUILD) {
-        applicationSupportDirectory += "/pr-binaries";
-    }
-    
+	QString applicationSupportDirectory = QCoreApplication::applicationDirPath();
+
```
Comment (AndrewMeadows) [inline stack-manager/src/GlobalData.cpp]: Indentation is bad.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ca8cd13c4876775843ee941d2b057e5f4fd9c084 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ca8cd13c4876775843ee941d2b057e5f4fd9c084 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f5dfc7d3e439d34a433477723c3106160f80e8ce available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c786789c4cd11996b63bd4cea2ead9eabaf05597 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 252086ea483f32f5097170c9fe11ec3abe2e975c available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by murillodigital** [comment]
@PhilipRosedale please try this installer:

https://s3.amazonaws.com/hifi-public/testPR/High+Fidelity-PR-6497-win32.exe

**Comment by hifi-gustavo** [comment]
Build console output for commit: eeebe0622851ee5e6090143e66b728838524527c available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cc6d30fe6f867080e61901588a023add803fecb7 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

File: interface/CMakeLists.txt [hunk 3]
```diff
@@ -201,7 +201,6 @@ else (APPLE)
 
   # link target to external libraries
   if (WIN32)
-  #  target_link_libraries(${TARGET_NAME} wsock32.lib Winmm.lib)
     target_link_libraries(${TARGET_NAME} wsock32.lib Winmm.lib)
```
Comment (birarda) [inline interface/CMakeLists.txt]: What makes this needed now?

Comment (murillodigital) [inline interface/CMakeLists.txt]: oh that's probably a left over test, I'll comment back

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9062cca8c474d83f33bc6fecfcd0a5c2e7e583ff available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9062cca8c474d83f33bc6fecfcd0a5c2e7e583ff available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dda99bfea65f561e1a7b5b7349d337e1a7eeda2e available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9ba5670c7611f8c09ff44bf47658778c735018e8 available at: https://gist.github.com/0eecd9172371e0584586<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6497.dmg

**Pull request merged**